str0m/src/session.rs

```diff
@@ -416,8 +416,8 @@ impl Session {
             if rid.is_none() && repaired_source.rid().is_some() {
                 rid = repaired_source.rid();
             }
+            let orig_seq_no = repaired_source.update(now, &header, clock_rate);
             let source = m_line.get_or_create_source_rx(ssrc);
-            let orig_seq_no = source.update(now, &header, clock_rate);
```

Comment:
This is not right. It's the main source that has the register for what is received and not. So `update()` must be on the main source.

It also returns the lengthened `orig_seq_no`. However you can't lengthen with the repaired sequence. It must be the real source (given that we set `header.sequence_number = orig_seq_16` higher up).

Reply:
That's what's now happening, before it wasn't correctly doing it on the main source.

`ssrc` here is the value from the header, which for a repaired packet is going to be the repair stream. The non-RTX case is handled on line 359. `repaired_ssrc` is the SSRC stream for the stream we are repairing

Reply:
Ah. Ok. As you were. :)